nav2_behavior_tree/plugins/action/controller_selector_node.cpp

```diff
@@ -36,16 +36,25 @@ ControllerSelector::ControllerSelector(
 
   getInput("topic_name", topic_name_);
 
+  rclcpp::QoS qos(rclcpp::KeepLast(1));
+  qos.transient_local().reliable();
+
   controller_selector_sub_ = node_->create_subscription<std_msgs::msg::String>(
-    topic_name_, 1, std::bind(&ControllerSelector::callbackControllerSelect, this, _1));
+    topic_name_, qos, std::bind(&ControllerSelector::callbackControllerSelect, this, _1));
 }
 
 BT::NodeStatus ControllerSelector::tick()
 {
   rclcpp::spin_some(node_);
 
   if (last_selected_controller_.empty()) {
-    getInput("default_controller", last_selected_controller_);
+    std::string default_controller;
+    getInput("default_controller", default_controller);
+    if (default_controller.empty()) {
```

Comment:
A comment here about what this is implementing would be great. Something just letting people know its so that if no default is set & none received failing. If you want to proceed without a set, must set a default. Would also be good context to add to the configuration page for this variable.